lab-david/__test__/auth-router.test.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const request = require('superagent');
+const User = require('../model/user.js');
+const serverToggle = require('../lib/server-toggle.js');
+const server = require('../server.js');
+
+require('jest');
+const url = `http://localhost:3000`;
+
+const exampleUser = {
+  username: 'exampleuser',
+  password: '1234',
+  email: 'exampleuser@test.com',
+};
+
+describe('Auth Routes', function(){
+  beforeAll( done => {
+    serverToggle.serverOn(server, done);
+  });
+
+  afterAll( done => {
+    serverToggle.serverOff(server, done);
+  });
+
+  describe('POST: /api/singup', function(){
+    describe('with a valid body', function(){
+      afterEach( done => {
+        User.remove({})
+          .then( () => done())
+          .catch(done);
+      });
+
+      it('should return a token', done => {
+        request.post(`${url}/api/signup`)
+          .send(exampleUser)
+          .end((err, res) => {
+            if(err) return done(err);
+            expect(res.status).toEqual(200);
+            expect(typeof res.text).toEqual('string');
+            done();
+          });
+      });
+    });
+
+    describe('with an invalid body', function(){
+      it('should return a 400 error', done => {
+        request.post(`${url}/api/signup`)
+          .send({})
+          .end((err, res) => {
+            expect(err.status).toEqual(400);
+            expect(res.status).toEqual(400);
+            done();
+          });
+      });
+    });
+  });
+
+  describe('GET: /api/signin', function(){
+    describe('with a valid body', function(){
+      beforeEach( done => {
```

Comment:
Consider building a mock module that will create your models for you, so you don't have to clog your tests files with them. You can then just import the module and call it to do the work for you. Cuts down on code too, DRY is good.